projects/plugins/boost/app/modules/cache/Boost_Cache.php

```diff
@@ -254,4 +272,208 @@ public function is_backend() {
 
 		return $is_backend;
 	}
+
+	/*
+	 * Checks if the post type is public.
+	 *
+	 * @param WP_Post $post - The post to check.
+	 * @return bool - True if the post type is public.
+	 */
+	protected function is_visible_post_type( $post ) {
+		$post_type = is_object( $post ) ? get_post_type_object( $post->post_type ) : null;
+		if ( empty( $post_type ) || ! $post_type->public ) {
+			return false;
+		}
+		return true;
+	}
+
+	/*
+	 * Delete the cache for the front page and paged archives.
+	 * This is called when a post is edited, deleted, or published.
+	 *
+	 * @param int $post_id - The id of the post that was edited, deleted, or published.
+	 */
+	protected function delete_front_page_cache( $post ) {
+		$front_page_id = get_option( 'show_on_front' ); // posts page
+		if ( $front_page_id === 'page' ) {
+			$front_page_id = get_option( 'page_on_front' ); // static page
+			if ( $front_page_id === $post->ID ) {
+				$this->delete_cache_for_post( $post->ID );
+			}
+		} else {
+			$this->storage->invalidate_home_page();
+			error_log( 'delete front page cache' ); // phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log
```

Comment:
I'm ok for the `error_log` statements to remain in here, pending being turned into proper logging statements as Adnan creates the logging system.